8081: search on pk and sk

kkoskelin opened the pull request:
ensure the records we are retrieving are user records and not user/case mappings

kkoskelin committed:
8081: search on pk and sk

ensure the records we are retrieving are user records and not user/case mappings

kkoskelin committed:
8081: don't use es6 style

rachaelparris commented:
Is there any type of test that could be written to cover this?

kkoskelin commented:
> Is there any type of test that could be written to cover this?

@rachaelparris I am open to suggestions on how to better test persistence methods connected to ES.  In the past there have been tests which basically say, "Make sure the search client gets JSON that looks like [this]" but it's not clear to me whether those tests actually have any value.

rachaelparris commented:
> > Is there any type of test that could be written to cover this?
> 
> @rachaelparris I am open to suggestions on how to better test persistence methods connected to ES. In the past there have been tests which basically say, "Make sure the search client gets JSON that looks like [this]" but it's not clear to me whether those tests actually have any value.

Maybe an integration test that performs a practitioner search that would have previously returned duplicates, and expect that only a single result is returned for each practitioner? There may even be an existing one you could just expand on to ensure there aren't duplicates.

kkoskelin committed:
8081: integration test to detect duplicates

Without additional check of pk and sk, test yields:
`
    Received: true

      50 |       `searchResults.${ADVANCED_SEARCH_TABS.PRACTITIONER}`,
      51 |     );
    > 52 |     expect(practitionerResultsContainDuplicates(searchResults)).toBeFalsy();
`
